examples/notes/shared/src/app.rs

```diff
@@ -778,85 +743,78 @@ mod save_load_tests {
+            let app = NoteEditor::default();
             let model = Default::default();
 
             Self {
                 app,
                 model,
                 subscription: None,
+                command: None,
                 edits: VecDeque::new(),
             }
         }
 
         // Update, picking out and keeping PubSub effects
-        fn update(&mut self, event: Event) -> (Vec<Effect>, Vec<Event>) {
-            let update = self.app.update(event, &mut self.model);
-
-            let mut effects = Vec::new();
-            let events = update.events.clone();
+        fn update(&mut self, event: Event) {
+            let mut cmd = self.app.update(event, &mut self.model);
 
-            for effect in update.into_effects() {
+            let mut subscribe = false;
+            for effect in cmd.effects() {
                 match effect {
                     Effect::PubSub(request) => match request.operation {
                         PubSubOperation::Subscribe => {
                             self.subscription = Some(request);
+                            subscribe = true;
                         }
                         PubSubOperation::Publish(bytes) => {
                             self.edits.push_back(bytes.clone());
                         }
                     },
-                    ef => effects.push(ef),
+                    _ => (),
                 }
             }
-            (effects, events)
+            if subscribe {
+                self.command = Some(cmd);
+            }
         }
 
         fn view(&self) -> ViewModel {
             self.app.view(&self.model)
         }
 
-        fn send_edits(&mut self, edits: &[Vec<u8>]) -> (Vec<Effect>, Vec<Event>) {
-            let subscription = self.subscription.as_mut().expect("to have a subscription");
-
-            let mut effects = Vec::new();
-            let mut events = Vec::new();
-
-            let evs = edits
-                .iter()
-                .flat_map(|ed| {
-                    self.app
-                        .resolve(subscription, Message(ed.clone()))
-                        .expect("should resolve")
-                        .events
-                })
-                .collect::<Vec<_>>();
-
-            for event in evs {
-                let (mut eff, mut ev) = self.update(event);
-
-                effects.append(&mut eff);
-                events.append(&mut ev);
+        fn send_edits(&mut self, edits: &[Vec<u8>]) {
+            for edit in edits {
+                print!("Sending edit: {:?}", edit);
+                self.subscription
+                    .as_mut()
+                    .expect("to have a subscription")
+                    .resolve(Message(edit.clone()))
+                    .expect("should resolve");
+
+                if let Some(cmd) = self.command.as_mut() {
+                    for event in cmd.events().collect::<Vec<_>>() {
+                        self.update(event.clone());
+                    }
```

Comment:
Ah, this is quite interesting! I see why you need to keep the original command.

Are you missing handling for any follow up commands coming out of the above `update` call?